music21/musicxml/m21ToXml.py

```diff
@@ -4421,21 +4441,39 @@ def noteToNotations(self, n, noteIndexInChord=0, chordParent=None):
             # get expressions from first note of chord
             chordOrNote = chordParent
 
-        # only apply expressions to notes or the first note of a chord...
-        if isSingleNoteOrFirstInChord:
-            for expObj in chordOrNote.expressions:
-                mxExpression = self.expressionToXml(expObj)
-                if mxExpression is None:
-                    # print('Could not convert expression: ', mxExpression)
-                    # TODO: should not!
-                    continue
-                if 'Ornament' in expObj.classes:
-                    if mxOrnaments is None:
-                        mxOrnaments = Element('ornaments')
-                    mxOrnaments.append(mxExpression)
-                    # print(mxExpression)
+        # apply all expressions apart from arpeggios only to the first note of a chord.
+        foundTopAndBottom = False
+        for expObj in chordOrNote.expressions:
+            mxExpression = None
+            if isinstance(expObj, expressions.ArpeggioMark):
+                if expObj.type == 'non-arpeggio':
+                    # <non-arpeggiate> goes on top and bottom note in chord
+                    if not foundTopAndBottom:
+                        topNoteNumber, bottomNoteNumber = topAndBottomIndices(chordOrNote)
+                        foundTopAndBottom = True
+                    if bottomNoteNumber == noteIndexInChord:
+                        mxExpression = self.expressionToXml(expObj)
+                        mxExpression.set('type', 'bottom')
+                    elif topNoteNumber == noteIndexInChord:
+                        mxExpression = self.expressionToXml(expObj)
+                        mxExpression.set('type', 'top')
                 else:
-                    notations.append(mxExpression)
+                    # <arpeggiate> goes on every note in the chord
+                    mxExpression = self.expressionToXml(expObj)
+            elif isSingleNoteOrFirstInChord:
+                mxExpression = self.expressionToXml(expObj)
+
+            if mxExpression is None:
+                # no expression is applicable, or expressionToXml() didn't recognize it.
+                continue
+
+            if 'Ornament' in expObj.classes:
+                if mxOrnaments is None:
+                    mxOrnaments = Element('ornaments')
+                mxOrnaments.append(mxExpression)
+                # print(mxExpression)
+            else:
+                notations.append(mxExpression)
```

Comment:
Seems like this creates a behavior change: now exporting ornaments stored on notes that are not first in the chord. We should revert or discuss it separately if you have a use case (see #909)

Reply:
Ooh!  Good eye.  I have a fix.